src/libs/actions/PersonalDetails.js

```diff
@@ -125,7 +112,7 @@ function fetch() {
     })
         .then((data) => {
             const allPersonalDetails = formatPersonalDetails(data.personalDetailsList);
-            Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, allPersonalDetails);
+            Onyx.set(ONYXKEYS.PERSONAL_DETAILS, allPersonalDetails);
```

Comment:
Why the change from `merge()` to `set()`?

Reply:
I think there may be some personal details created that are not in the `personalDetailsList` e.g. those added when loading reports or creating a chat with a brand new user. Is there a chance we are overwriting those here?

Reply:
Before we needed to `merge` otherwise I believe we'd overwrite the `timeZone` that was added to the personalDetails when we called `fetchTimezone()` but now the details returned from `personalDetailsList` and `PersonalDetails_GetForEmails` both contain the timeZone so there's no need to merge instead of set. 

> Is there a chance we are overwriting those here?

I suppose if we're getting different persoanlDetaials from the report, then yes we would be overwriting it, so it may be best to stick with `merge` over `set` in the end.

Reply:
Ok cool. I think that's probably a safe bet here. It sort of makes sense to favor using `merge()` over `set()` unless there is a specific reason why we might want to blow away everything else we've got.